packages/tools/src/tools/OverlayGridTool.ts

```diff
@@ -0,0 +1,384 @@
+    const {
+      imagePositionPatient,
+      rows,
+      columns,
+      rowCosines,
+      columnCosines,
+      rowPixelSpacing,
+      columnPixelSpacing,
+    } = metaData.get('imagePlaneModule', imageId);
+
+    // top left world, top right world, bottom right world, bottom left world
+    const topLeft = <Types.Point3>[...imagePositionPatient];
+    const topRight = <Types.Point3>[...imagePositionPatient];
+    const bottomLeft = <Types.Point3>[...imagePositionPatient];
+    const bottomRight = <Types.Point3>[...imagePositionPatient];
+
+    vec3.scaleAndAdd(
+      topRight,
+      imagePositionPatient,
+      columnCosines,
+      columns * columnPixelSpacing
+    );
+    vec3.scaleAndAdd(
+      bottomLeft,
+      imagePositionPatient,
+      rowCosines,
+      rows * rowPixelSpacing
+    );
+
+    vec3.scaleAndAdd(
+      bottomRight,
+      bottomLeft,
+      columnCosines,
+      columns * columnPixelSpacing
+    );
+
+    // check if the topLeft and bottomLeft line is parallel to the viewUp
+    const pointSet1 = [topLeft, bottomLeft, topRight, bottomRight];
+    const pointSet2 = [topLeft, topRight, bottomLeft, bottomRight];
+
+    return { pointSet1, pointSet2 };
+  };
+  /**
+   * it is used to draw the length annotation in each
+   * request animation frame. It calculates the updated cached statistics if
+   * data is invalidated and cache it.
+   *
+   * @param enabledElement - The Cornerstone's enabledElement.
+   * @param svgDrawingHelper - The svgDrawingHelper providing the context for drawing.
+   */
+  renderAnnotation = (
+    enabledElement: Types.IEnabledElement,
+    svgDrawingHelper: SVGDrawingHelper
+  ): boolean => {
+    let renderStatus = false;
+    const { viewport: targetViewport } = enabledElement;
+
+    const targetImageIds = targetViewport.getImageIds();
+    if (targetImageIds.length < 2) {
+      return renderStatus;
+    }
+
+    const { annotations } = this.editData;
+    if (!annotations) {
+      return renderStatus;
+    }
+
+    [...annotations.values()].forEach((annotation) => {
+      const sourceViewport = annotation.data.sourceViewport;
+      if (!sourceViewport) {
+        return renderStatus;
+      }
+
+      if (sourceViewport.id === targetViewport.id) {
+        // If the source viewport is the same as the current viewport, we don't need to render
+        return renderStatus;
+      }
+
+      if (
+        sourceViewport.getFrameOfReferenceUID() !==
+        targetViewport.getFrameOfReferenceUID()
+      ) {
+        return renderStatus;
+      }
+
+      if (!annotation) {
+        return renderStatus;
+      }
+
+      const styleSpecifier: StyleSpecifier = {
+        toolGroupId: this.toolGroupId,
+        toolName: this.getToolName(),
+        viewportId: enabledElement.viewport.id,
+      };
+
+      const { focalPoint, viewPlaneNormal } = targetViewport.getCamera();
+      const { viewPlaneNormal: sourceViewPlaneNormal } =
+        sourceViewport.getCamera();
+
```

Comment:
There is no source viewport, it should be based on the sourceImagIds, and you can calculate this based on the metadata (cosines)